### PR TITLE
Win CI: Enable more specs, add zlib and yaml

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

File: src/crystal/system/win32/file.cr [hunk 1]
```diff
@@ -26,8 +26,8 @@ module Crystal::System::File
     fd
   end
 
-  def self.mktemp(prefix : String, suffix : String?, dir : String) : {LibC::Int, String}
-    path = "#{tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
+  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
+    path = "#{Crystal::System::Dir.tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
```
Comment (Sija) [inline src/crystal/system/win32/file.cr:30]: Shouldn't this be similar to the unix impl? 👇 https://github.com/crystal-lang/crystal/blob/625884b1acbbbf4db321ac76738ebef35a0aba93/src/crystal/system/unix/file.cr#L16-L18

```suggestion
  def self.mktemp(prefix, suffix, dir) : {LibC::Int, String}
    path = "#{dir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Comment (straight-shoota) [reply src/crystal/system/win32/file.cr:30]: The method signature doesn't matter much, but yes, the `dir` argument should be honoured.

File: spec/win32_std_spec.cr [hunk 2]
```diff
@@ -1,64 +1,75 @@
...
 # require "./std/big/big_float_spec.cr" (failed linking)
 # require "./std/big/big_int_spec.cr" (failed linking)
 # require "./std/big/big_rational_spec.cr" (failed linking)
-# require "./std/big/number_spec.cr" (failed codegen)
+# require "./std/big/number_spec.cr" (failed linking)
 require "./std/bit_array_spec.cr"
 require "./std/bool_spec.cr"
 require "./std/box_spec.cr"
-# require "./std/callstack_spec.cr" (failed codegen)
+# require "./std/callstack_spec.cr"
```
Comment (RX14) [inline spec/win32_std_spec.cr:17]: What happened here?

Comment (oprypin) [reply spec/win32_std_spec.cr:17]: Yes, it is neither of those. It no longer fails codegen and it doesn't crash, but it just has a lot of test failures that I don't want to deal with as part of this PR. So I commented it out.

**Comment by oprypin** [comment]
Why does it dismiss reviews now? There was no force-push 😩

**Comment by straight-shoota** [comment]
By adding the merge commit the PR head changed.

**Comment by oprypin** [comment]
I can get that, but it should be allowed still

**Comment by bcardiff** [comment]
We changed that repo setting to avoid having superfluous/old approvals @oprypin .

File: spec/win32_std_spec.cr [hunk 3]
```diff
@@ -1,64 +1,75 @@
...
 require "./std/comparable_spec.cr"
 require "./std/complex_spec.cr"
+require "./std/compress/deflate/deflate_spec.cr"
+require "./std/compress/gzip/gzip_spec.cr"
+require "./std/compress/zip/zip_file_spec.cr"
+require "./std/compress/zip/zip_spec.cr"
+require "./std/compress/zlib/reader_spec.cr"
+require "./std/compress/zlib/stress_spec.cr"
+require "./std/compress/zlib/writer_spec.cr"
+# require "./std/concurrent/future_spec.cr" (failed codegen)
```
Comment (bcardiff) [inline spec/win32_std_spec.cr:32]: FYI this file no longer exists on master.

File: spec/win32_std_spec.cr (hunk 2 above)
Comment (bcardiff) [reply spec/win32_std_spec.cr:17]: This file has moved to call_stack_spec on master, which is already included later.

File: src/crystal/system/win32/file.cr [hunk 4]
```diff
@@ -26,8 +26,9 @@ module Crystal::System::File
     fd
   end
 
-  def self.mktemp(prefix : String, suffix : String?, dir : String) : {LibC::Int, String}
-    path = "#{tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
+  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
```
Comment (bcardiff) [inline src/crystal/system/win32/file.cr:29]: It's odd that these type restrictions are not present in the unix counter part

File: spec/win32_std_spec.cr (hunk 3 above)
Comment (oprypin) [reply spec/win32_std_spec.cr:32]: Thanks. I double-checked by re-generating the list, and your two comments cover everything.

File: src/crystal/system/win32/file.cr (hunk 4 above)
Comment (oprypin) [reply src/crystal/system/win32/file.cr:29]: I suppose, yes. I'm not changing the status quo, though.

**Comment by oprypin** [comment]
It dismisses reviews on *any* commit??? That's just unacceptable. Please undo that setting and let the person who merges judge what's stale or not. It's hard enough to get 2 people to approve as is.
In this particular case, if @bcardiff approved and I made only a tiny change exactly in accordance to his comment, obviously that's not stale.

**Comment by straight-shoota** [comment]
@oprypin Let's focus on the PR here. Meta discussion about GitHub settings should be separate.

**Comment by oprypin** [comment]
Rest assured that the thought crossed my mind, but I don't know what the place would be.
And feel free to collapse the comments.

File: src/crystal/system/win32/file.cr [hunk 5]
```diff
@@ -26,8 +26,9 @@ module Crystal::System::File
...
+    dir = dir + ::File::SEPARATOR
+    path = "#{dir}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```
Comment (straight-shoota) [inline src/crystal/system/win32/file.cr:31]: This avoids an extra allocation:
```suggestion
    path = "#{dir}#{::File::SEPARATOR}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```
Or use `::File.join` I think we can depend on that from here.

Comment (oprypin) [reply src/crystal/system/win32/file.cr:31]: This is not by accident, it's just copying the implementation from unix. I'd prefer to not refactor that here (pretend that I'm "not changing the code"); maybe someone can do both at the same time.

```
$ grep -rF1 'dir = dir + ::File::SEPARATOR'
src/crystal/system/unix/file.cr-  def self.mktemp(prefix, suffix, dir) : {LibC::Int, String}
src/crystal/system/unix/file.cr:    dir = dir + ::File::SEPARATOR
src/crystal/system/unix/file.cr-    path = "#{dir}#{prefix}.XXXXXX#{suffix}"
--
src/crystal/system/win32/file.cr-  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
src/crystal/system/win32/file.cr:    dir = dir + ::File::SEPARATOR
src/crystal/system/win32/file.cr-    path = "#{dir}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Comment (oprypin) [reply src/crystal/system/win32/file.cr]: Ah whatever, applied it. Thanks. Leaving it wouldn't make as much sense as I make it seem.

**Pull request merged**